hexrd/material.py

```diff
@@ -199,6 +191,34 @@ def __str__(self):
         s += '   plane Data:  %s' % str(self.planeData)
         return s
 
+    def _newUnitcell(self):
+        """
+        @author Saransh Singh, Lawrence Livermore National Lab
+        @date 03/11/2021 SS 1.0 original
+        @details create a new unitcell class with everything initialized
+        correctly
+        """
+        lparms = [x.value for x in self._lparms]
+        ltype = symmetry.latticeType(self.sgnum)
+        lparms = unitcell._rqpDict[ltype][1](lparms)
+        lparms_vu = []
+        for i in range(6):
+            if(i < 3):
+                lparms_vu.append(_angstroms(lparms[i]))
+            else:
+                lparms_vu.append(_degrees(lparms[i]))
+
+        self._lparms = lparms_vu
```

Comment:
It might be nice to put this section into a separate function for clarity, such as `_reset_lparms()` or `_sgnum_changed()`.

Reply:
I second this.

Reply:
Gotcha!

Reply:
I can just update the lattice parameter and then set the sgnum since that is a property in unitcell and updates everything. But I think it'll be the same computational time as initializing a new `unitcell` since each function in the `__init__` is pretty much called again. Need to check that though.